src/bindings/io/doc/io_doc.c

```diff
@@ -0,0 +1,365 @@
+	void * data;
+} SomeIoLibHandle;
+
+// Example I/O mangement library bitmask flags
+typedef enum {
+	SOME_IOLIB_READABLE = 1 << 0,
+	SOME_IOLIB_WRITABLE = 1 << 1,
+} SomeIoLibFlags;
+
+/**
+ * Container for required additional information for
+ * IO binding operations
+ *
+ * It is helpful to create a data structure for your binding to store additional data
+ */
+typedef struct DocOperationData
+{
+	union {
+		ElektraIoFdOperation * fd;
+		ElektraIoTimerOperation * timer;
+		ElektraIoIdleOperation * idle;
+	} operation;
+	// Add additional fields as required
+	char * bar;
+} DocOperationData;
+
+typedef struct DocBindingData
+{
+	// Add additional fields as required
+	char * foo;
+} DocBindingData;
+
+/**
+ * A little helper function for allocating our custom operation data structure
+ * @return  New data structure, remember to call elektraFree()
+ */
+static DocOperationData * newOperationData (void)
+{
+	DocOperationData * operationData = elektraMalloc (sizeof (*operationData));
+	if (operationData == NULL)
+	{
+		ELEKTRA_LOG_WARNING ("elektraMalloc failed");
+		return NULL;
+	}
+	memset (operationData, 0, sizeof (*operationData));
+
+	return operationData;
+}
+
+/**
+ * Convert your I/O library bit mask to Elektra's IO flags
+ * @param  bitmask   bit mask from I/O management library
+ * @return           bit mask with Elektra's IO flags
+ */
+static int someBitMaskToElektraIoFlags (int bitmask)
+{
+	int flags = 0;
+	if (bitmask & SOME_IOLIB_READABLE)
+	{
+		flags |= ELEKTRA_IO_READABLE;
+	}
+	if (bitmask & SOME_IOLIB_READABLE)
+	{
+		flags |= ELEKTRA_IO_WRITABLE;
+	}
+	return flags;
+}
+
+/**
+ * Calls the associated operation callback
+ * Called by your I/O management library whenever a file descriptor status has changed.
+ *
+ * @param handle some I/O management handle
+ * @param flags flags bit mask (already in Elektra's format)
+ */
+static void ioDocBindingFdCallback (SomeIoLibHandle * handle, int bitmask)
+{
+	// For this example let's assume handle was passed as argument
+	assert (handle->data != NULL);
+	DocOperationData * operationData = (DocOperationData *)handle->data;
+	ElektraIoFdOperation * fdOp = (ElektraIoFdOperation *)operationData->operation.fd;
+
+	// Flag bit mask will not match ELEKTRA_IO_WRITABLE or ELEKTRA_IO_READABLE values
+	elektraIoFdGetCallback (fdOp) (fdOp, someBitMaskToElektraIoFlags (bitmask));
+}
+
+/**
+ * Calls the associated operation callback
+ * Called by your I/O management library whenever a timer interval has passed.
+ *
+ * @param handle some I/O management handle
+ */
+static void ioDocBindingTimerCallback (SomeIoLibHandle * handle)
+{
+	// For this example let's assume handle was passed as argument
+	assert (handle->data != NULL);
+	DocOperationData * operationData = (DocOperationData *)handle->data;
+	ElektraIoTimerOperation * timerOp = (ElektraIoTimerOperation *)operationData->operation.timer;
+
+	elektraIoTimerGetCallback (timerOp) (timerOp);
+}
+
+/**
+ * Calls the associated operation callback
```

Comment:
A dot at the end might be needed so that Doxygen correctly knows what `@brief` is. Or are both lines in brief on purpose?